tests/components/mqtt/test_sensor.py

```diff
@@ -292,7 +292,6 @@ async def test_setting_sensor_value_via_mqtt_json_message(
                 sensor.DOMAIN: {
                     "name": "test",
                     "state_topic": "test-topic",
-                    "unit_of_measurement": "fav unit",
```

Comment:
I believe that `device_class` can be `None`. Then cannot `unit_of_measurement` be set to anything custom?

Reply:
uom can be anything custom, it cannot be custom when using a device class  (so it is the other way around Jan 😉 )

Reply:
Good to known